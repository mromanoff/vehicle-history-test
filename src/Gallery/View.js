import React from 'react';
import Loader from '../components/Loader';
import Error from '../components/Error';
import Pagination from './Pagination';
import CategoryPicker from './CategoryPicker';
import Photo from './Photo';

const View = ({
  isLoading,
  isError,
  photos,
  error,
  categories,
  currentCategory,
  onCategoryChange,
  onPageChange,
  currentPage,
  totalItemsCount,
  itemsCountPerPage,
  RenderPagination,
  RenderPhoto,
  RenderCategoryPicker,
  RenderLoader,
  RenderError,
}) => (
  <div>
    <RenderCategoryPicker
      categories={categories}
      currentCategory={currentCategory}
      onCategoryChange={onCategoryChange}
    />
    <section>
      {isLoading && <RenderLoader />}
      {isError && <RenderError error={error} />}

      <div className={isLoading ? 'loading' : 'loaded'}>
        {photos && (
          <div>

            <RenderPagination
              currentPage={currentPage}
              totalItemsCount={totalItemsCount}
              itemsCountPerPage={itemsCountPerPage}
              onChange={onPageChange}
            />

            <div className="module gallery">
              {photos.map(photo => (
                <RenderPhoto key={photo.id} photo={photo} />
              ))}
            </div>

            <RenderPagination
              currentPage={currentPage}
              totalItemsCount={totalItemsCount}
              itemsCountPerPage={itemsCountPerPage}
              onChange={onPageChange}
            />

          </div>
        )}
      </div>
    </section>
  </div>
);
View.propTypes = {};

View.defaultProps = {
  RenderCategoryPicker: CategoryPicker,
  RenderPhoto: Photo,
  RenderPagination: Pagination,
  RenderLoader: Loader,
  RenderError: Error,
};

export default View;
