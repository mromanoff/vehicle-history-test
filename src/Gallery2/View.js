import React from 'react';
import Loader from '../components/Loader';
import Error from '../components/Error';
import Pagination from '../components/Pagination';
import CategoryPicker from './CategoryPicker';
import Photo from './Photo';

const View = ({
  isLoading,
  error,
  data,
  categories,
  initialCategory,
  initialPage,
  onLoad,
  RenderPagination,
  RenderPhoto,
  RenderCategoryPicker,
  RenderLoader,
  RenderError,
}) => (
  <div>
    <RenderCategoryPicker
      categories={categories}
      initialCategory={initialCategory}
      onCategoryPick={onLoad}
    />
    <section>
      {isLoading && <RenderLoader />}
      {error && <RenderError error={error} />}

      <div className={isLoading ? 'loading' : 'loaded'}>
        <RenderPagination initialPage={initialPage} />

        {data && (
          <div className="module gallery">
            {data.photos.map(photo => (
              <RenderPhoto key={photo.id} photo={photo} />
            ))}
          </div>
        )}
      </div>
    </section>
  </div>
);
View.propTypes = {};

View.defaultProps = {
  RenderCategoryPicker: CategoryPicker,
  RenderPhoto: Photo,
  RenderPagination: Pagination,
  RenderLoader: Loader,
  RenderError: Error,
};

export default View;
