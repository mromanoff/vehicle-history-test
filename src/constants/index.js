export const CATEGORIES = ['Modeling', 'Animals', 'Transportation', 'Nature', 'Sport'];
