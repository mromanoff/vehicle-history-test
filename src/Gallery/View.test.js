import React from 'react';
import View from './View';
import { CATEGORIES } from '../constants/index';

const photos = [{ id: 1 }];

it('renders view', () => {
  const wrapper = shallow(
    <View
      categories={CATEGORIES}
      currentCategory={CATEGORIES[0]}
      onCategoryChange={() => {}}
      currentPage={1}
      itemsCountPerPage={25}
      totalItemsCount={100}
      photos={photos}
      onPageChange={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders loading view', () => {
  const wrapper = shallow(
    <View
      isLoading={true}
      categories={CATEGORIES}
      currentCategory={CATEGORIES[0]}
      onCategoryChange={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders error view', () => {
  const wrapper = shallow(
    <View
      isError={true}
      categories={CATEGORIES}
      currentCategory={CATEGORIES[0]}
      onCategoryChange={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});