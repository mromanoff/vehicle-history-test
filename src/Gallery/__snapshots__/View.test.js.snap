// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error view 1`] = `
<div>
  <CategoryPicker
    categories={
      Array [
        "Modeling",
        "Animals",
        "Transportation",
        "Nature",
        "Sport",
      ]
    }
    currentCategory="Modeling"
    onCategoryChange={[Function]}
  />
  <section>
    <LoadError />
    <div
      className="loaded"
    />
  </section>
</div>
`;

exports[`renders loading view 1`] = `
<div>
  <CategoryPicker
    categories={
      Array [
        "Modeling",
        "Animals",
        "Transportation",
        "Nature",
        "Sport",
      ]
    }
    currentCategory="Modeling"
    onCategoryChange={[Function]}
  />
  <section>
    <Loader />
    <div
      className="loading"
    />
  </section>
</div>
`;

exports[`renders view 1`] = `
<div>
  <CategoryPicker
    categories={
      Array [
        "Modeling",
        "Animals",
        "Transportation",
        "Nature",
        "Sport",
      ]
    }
    currentCategory="Modeling"
    onCategoryChange={[Function]}
  />
  <section>
    <div
      className="loaded"
    >
      <div>
        <GalleryPagination
          currentPage={1}
          itemsCountPerPage={25}
          onChange={[Function]}
          totalItemsCount={100}
        />
        <div
          className="module gallery"
        >
          <Photo
            key="1"
            photo={
              Object {
                "id": 1,
              }
            }
          />
        </div>
        <GalleryPagination
          currentPage={1}
          itemsCountPerPage={25}
          onChange={[Function]}
          totalItemsCount={100}
        />
      </div>
    </div>
  </section>
</div>
`;
